### ISSUE TITLE
Doesn't pick the correct cards on turn 3

### ISSUE THREAD

**Issue by d7c0** [Fate-Grand-Automata/FGA#1861]
### Preparation

- [X] I tested the [latest release](https://github.com/Fate-Grand-Automata/FGA/releases/tag/canary)
- [X] I looked at other issues (even the [closed ones](https://github.com/Fate-Grand-Automata/FGA/issues?q=is%3Aissue+is%3Aclosed))
- [X] I read the [Troubleshooting Guide](https://github.com/Fate-Grand-Automata/FGA/wiki/Troubleshooting)

### FGO server

NA

### FGA build number

2690

### Describe the bug

FGA Ignores card priority and servant priority on turn 3.

![image](https://github.com/user-attachments/assets/d50de611-82a3-4ac5-86d8-1f919435232f)

### Video

https://github.com/user-attachments/assets/3679cc80-0bc0-47d9-8db4-6bf6f625ec04

### Device model

LDPlayer9 "SM-G9880"

### Android version

7

**Comment by vybze** [comment]
Are you using spam? Card priorit, servant priority and spam option doesn't work well together

**Comment by d7c0** [comment]
> Are you using spam? Card priorit, servant priority and spam option doesn't work well together

Yes, I'm using spam from turn 3 and onwards. I suppose that's most likely it, when combined with an NP, but it still seems sort of random.

**Comment by andydabin** [comment]
I am having the same issue, and I have spam turned off. It very consistently chooses spishtar's cards over morgans cards only on wave 3 (although, it seems to pick the correct priority order on turn 7 weirdly enough). This bug happens every run on both my bluestacks emulator and my galaxy s23 plus phone, and has also been verified by others to happen with other servants as well.

**Comment by andydabin** [comment]
https://www.youtube.com/watch?v=eQwUjVCpsXo

https://www.youtube.com/watch?v=O4VLRL1DqrI

https://www.youtube.com/watch?v=8ioVDMlh4J0

https://www.youtube.com/watch?v=CwwKzRx4RwY

**Comment by andydabin** [comment]
![Screenshot_20240815_043612_FGA](https://github.com/user-attachments/assets/b1c7dd43-4a1d-4f31-b373-9dea8fb230b3)
![Screenshot_20240815_043618_FGA](https://github.com/user-attachments/assets/d9543de3-026b-457c-81e4-c22b67631a94)

Same issue with both of these settings.
